src/ot_orchestration/operators/platform.py

```diff
@@ -0,0 +1,103 @@
+    """Custom operator that decides whether to run a PIS step or not.
+
+    At the moment, this operator will check the parts of the PIS configuration
+    that are relevant to the current step, comparing them against the copy in the
+    specified bucket. If they are different, that means the step should be
+    run.
+
+    It would be interesting add a check that downloads the manifest from the
+    specified bucket and compares the resources listed for the step with the
+    files in the bucket, along with their checksums. That way we can ensure the
+    files exist and have not been tampered with.
+
+    :param project_id: The GCP project ID. Defaults to the platform project.
+    :param step_name: The name of the PIS step to check.
+    :param local_config_path: The path to the local configuration file.
+    :param upstream_config_url: The URL of the upstream configuration file.
+    """
+
+    template_fields: Sequence[str] = (
+        "step_name",
+        "local_config_path",
+        "upstream_config_url",
+    )
+
+    def __init__(
+        self,
+        *args,
+        project_id: str = GCP_PROJECT_PLATFORM,
+        step_name: str,
+        local_config_path: Path,
+        upstream_config_url: str,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.step_name = step_name
+        self.project_id = project_id
+        self.local_config_path = local_config_path
+        self.upstream_config_url = upstream_config_url
+
+    def get_local_config(self) -> dict[str, Any]:
+        """Read the local configuration file and return the relevant part."""
+        return read_yaml_config(self.local_config_path)
+
+    def get_upstream_config(self, url: str) -> dict[str, Any]:
+        """Download the upstream configuration file and return the relevant part."""
+        c = Client(self.project_id)
+        b = Bucket(client=c, name=bucket_name(url))
+        blob = b.blob(bucket_path(url))
+        upstream_config = blob.download_as_bytes()
+        return yaml.safe_load(upstream_config)
+
+    def extract_relevant_config(self, config: dict[str, Any]) -> dict[str, Any]:
+        """Get the relevant part of the configuration for the current step."""
+        return {
+            "scratchpad": config.get("scratchpad", None),
+            "step": config.get("steps", {}).get(self.step_name, None),
+        }
+
+    def choose_branch(self, context: Context) -> str | Iterable[str]:
+        """Decide whether to run the current PIS step or not."""
+        local_config = self.get_local_config()
+        upstream_config = {}
+
+        try:
+            upstream_config = self.get_upstream_config(self.upstream_config_url)
+        except NotFound:
+            self.log.info(
+                "Upstream configuration file not found, assuming first run, %s",
+                self.upstream_config_url,
+            )
+
+        lc = self.extract_relevant_config(local_config)
```

Comment:
The step parameters change will not trigger the rerun in this context.

Reply:
Can you clarify this one? I don't understand what you mean.

Reply:
Imagine case when you just change the step config for instance from
```
    - name: download mondo
      source: https://github.com/monarch-initiative/mondo/releases/download/v2021-12-30/mondo.json
      destination: ontology-inputs/ontology-mondo.json
```
to 
```
    - name: download mondo
      source: https://github.com/monarch-initiative/mondo/releases/download/v2024-12-12/mondo.json
      destination: ontology-inputs/ontology-mondo.json
```
(change in the date of mondo source). 

Will it trigger the rerun in that context?

Reply:
It should, it gets both the scratchpad and the dict for the step that is running (Lines 72-75) and decides the branch based on whether any of those two dictionaries are different (line 96).

Am I missing something?

Reply:
my mistake, you are correct. I missed nested `get` when you extract the step config.